File: array_and_hashing/cpp/validAnagram.cpp
#include <map>
#include <string>

bool isAnagram(std::string s, std::string t) {
    if(s.size() != t.size()){
        return false;
    }
    std::map<char, int> sMap;
    std::map<char, int> tMap;

    for(int i = 0; i < s.size(); i++){
        sMap[s[i]]++;
        tMap[t[i]]++;
    }

    return sMap == tMap;
}

File: array_and_hashing/swift/validAnagram.swift
func isAnagram(_ s: String, _ t: String) -> Bool {
    if(s.count != t.count){
        return false
    }
    var s = Array(s), t = Array(t)
    var mappedS = s.map{($0, 1)}
    var mappedT = t.map{($0, 1)}
    return  Dictionary(mappedS, uniquingKeysWith: +) == Dictionary(mappedT, uniquingKeysWith: +)
}